src/Microsoft.Diagnostics.Monitoring.WebApi/Metrics/MetricsSettingsFactory.cs

```diff
@@ -61,6 +65,14 @@ public static MetricsPipelineSettings CreateSettings(GlobalCounterOptions counte
                 eventPipeCounterGroups.Add(new EventPipeCounterGroup { ProviderName = MonitoringSourceConfiguration.GrpcAspNetCoreServer, Type = CounterGroupType.EventCounter });
             }
 
+            foreach (EventPipeCounterGroup counterGroup in eventPipeCounterGroups)
+            {
+                if (intervalMap.TryGetValue(counterGroup.ProviderName, out GlobalProviderOptions providerInterval))
+                {
+                    counterGroup.IntervalSeconds = providerInterval.IntervalSeconds;
```

Comment:
Should this code throw if `counterGroup.IntervalSeconds` is not null AND does not match `providerInterval.IntervalSeconds` value? In addition to this code if `counterGroup.IntervalSeconds` was null.

Reply:
We have just constructed counterGroup, so it will always start as null.

Reply:
> We have just constructed counterGroup, so it will always start as null.

This function does not know that; that is only necessarily true for the default groups. The `createCounterGroups` callback can construct whatever it wants, including non-null values for this property.

If you don't want to do that, at least `Debug.Assert` that each of these are null.